src/run.rs

```diff
@@ -326,23 +328,50 @@ pub fn clean_up(
         }
     } else {
         let _ = fs::remove_file(delete_file);
+        let tmp_dir = preserve_dir(tmp_dir, project_name, preserved_project_dir)?;
 
-        let mut tmp_dir = tmp_dir.into_path();
+        log::info!("Project directory preserved at: {}", tmp_dir.display());
+    }
 
-        if let Some(name) = project_name {
-            let new_path = tmp_dir.with_file_name(name);
+    kill_subprocesses(subprocesses)?;
 
-            fs::rename(&tmp_dir, &new_path)?;
+    Ok(())
+}
 
-            tmp_dir = new_path;
+// preserve dir by renaming to appropriate name if project_name is set
+// and moves the project dir to preserved_project_dir as defined by the user
+// it returns the dir where the project is preserved
```

Comment:
I think you can remove this comments, the code is readable enough to be understood. What do you think?

Reply:
Removed :)